modules/controller/index.js

```diff
@@ -37,193 +159,91 @@ Controller.prototype._relations = function (request) {
   if (result && !Array.isArray(result)) {
     result = [result];
   }
-  return this.validRelations(_.uniq(result));
+  return _.intersection(this.source.relations(), _.uniq(result));
 };
 
-Controller.prototype.validFilters = function (filters) {
-  return _.intersection(this.source.filters(), filters);
-};
+Controller.prototype._readRelatedHandler =
+  function(opts, request, response, next) {
+    var source = this.source;
+    var relation = request.params ? request.params.relation : null;
+    var responder = opts.responder || this.responder;
 
-Controller.prototype.validRelations = function(relations) {
-  return _.intersection(this.source.relations(), relations);
-};
 
-Controller.prototype.responder = require('./lib/responder');
+    source.byId(request.params.id, relation).then(function (model) {
```

Comment:
I take this back, I we can totally specify a promise based API for the source and just promisify anything that isn't already.